ipn/ipnlocal/local.go

```diff
@@ -5476,6 +5481,46 @@ func (b *LocalBackend) DebugBreakDERPConns() error {
 	return b.magicConn().DebugBreakDERPConns()
 }
 
+func (b *LocalBackend) pushWebUIUpdateProgress(up ipnstate.UpdateProgress) {
+	b.mu.Lock()
+	defer b.mu.Unlock()
+	b.logf("update progress: %+v", up)
+	b.webUIUpdateProgress = append(b.webUIUpdateProgress, up)
+}
+
+func (b *LocalBackend) GetWebUIUpdateProgress() []ipnstate.UpdateProgress {
```

Comment:
is there anything about these methods that are unique to the WebUI, even if that's the primarily (and initially, the only) user of them?  If `tailscale update` eventually decided to use the localAPI calls, would they still be able to use these?

Reply:
it's mostly that only the web UI is using them right now. tbh the biggest problem I had was writing names for things, since it was hard to not conflate updates as in what this PR is doing, updates like node updates, updates like progress messages telling the client how far their first kind of update was done, etc...

happy to replace "webUIUpdate" with a more useful string, if you have ideas